src/Common/src/TypeSystem/Common/InstantiatedType.cs

```diff
@@ -159,7 +159,22 @@ public override MethodDesc GetFinalizer()
             MethodDesc typicalFinalizer = _typeDef.GetFinalizer();
             if (typicalFinalizer == null)
                 return null;
-            return _typeDef.Context.GetMethodForInstantiatedType(typicalFinalizer, this);
+
+            MetadataType typeInHierarchy = this;
+            while (typicalFinalizer.OwningType.GetTypeDefinition() != typeInHierarchy.GetTypeDefinition())
```

Comment:
Isn't the `.GetTypeDefinition()` part on `typicalFinalizer.OwningType.GetTypeDefinition()` a NOP?